src/gpp_client/rest.py

```diff
@@ -0,0 +1,108 @@
+import aiohttp
+import certifi
+import gzip
+import ssl
+
+
+__all__ = ["_GPPRESTClient"]
+
+
+class _GPPRESTClient:
+    """
+    REST API client to non-GraphQL requests that help with the function of managers and coordinators.
+
+    Attributes
+    ----------
+    base_url : str
+        Base URL of the REST API. Derived from GPPClient base_url.
+    gpp_token : str
+        GPP token to authenticate against the REST API. Same as GPPClient.
+    """
+
+    def __init__(self, resolved_url: str, gpp_token: str) -> None:
+        self.base_url = resolved_url.rstrip("/odb")
+        self.gpp_token = gpp_token
+        self._session = None
+
+    def resolve_headers(self):
+        headers = {
+            "Content-Type": "text/plain",
+            "Authorization": f"Bearer {self.gpp_token}",
+        }
+        return headers
+
+    async def _get_session(self):
+        if self._session is None or self._session.closed:
+            ssl_context = ssl.create_default_context(cafile=certifi.where())
+            connector = aiohttp.TCPConnector(ssl=ssl_context)
+            headers = self.resolve_headers()
+            self._session = aiohttp.ClientSession(
+                base_url=self.base_url,
+                timeout=aiohttp.ClientTimeout(total=30),
+                connector=connector,
+                headers=headers,
+            )
+        return self._session
+
```

Comment:
Please add an async function to close the session.

Reply:
Eventually we will want to use context managers to handle entering and exiting the session.